developers/contracts.mdx

```diff
@@ -6,22 +6,64 @@ For mev-commit testnet contract addresses please refer to [Testnet](/developers/
+
+
+- **Bidder Registry**
+
+  This contract manages the monetary aspects of the protocol. Bidders deposit funds into this contract to participate in the bidding process. The contract also handles the retrieval of funds by bidders and processes payments to providers. Events from the contract:
+    - BidderRegistered: Emitted when a new bidder registers with the protocol.
+    - FundsRetrieved: Emitted when a bidder successfully retrieves their funds.
+    - FundsRewarded: Emitted when funds are transferred as a reward to a provider.
+    - BidderWithdrawal: Emitted when a bidder withdraws their stake or funds from the contract.
+    - ProtocolFeeRecipientUpdated: Emitted when the recipient of protocol fees is updated.
+
+- **Provider Registry**
+  
+  This contract stores the stake that providers bind to ensure they fulfill their commitments under the preconfirmation process. It handles the registration of providers, the deposit of funds, slashing for failing to meet commitments, and the withdrawal of funds. Events from the contract:
+    - ProviderRegistered: Emitted when a new provider registers with the protocol.
+    - FundsDeposited: Emitted when a provider deposits funds into the contract as a stake.
+    - FundsSlashed: Emitted when a provider’s funds are slashed due to a failure to meet their commitments.
+    - Unstake: Emitted when a provider begins the unstaking process.
+    - Withdraw: Emitted when a provider withdraws their funds after unstaking.
+    - WithdrawalDelayUpdated: Emitted when the delay period for withdrawals is updated.
+    - PenaltyFeeRecipientUpdated: Emitted when the recipient of penalty fees is updated.
+    - FeePayoutPeriodBlocksUpdated: Emitted when the payout period for fees in terms of block count is updated.
+
+- **Oracle**
+
+    The Oracle contract retrieves and processes details about which commitments should be rewarded or slashed. Access to its functions is restricted to the owner of the Oracle     key, ensuring that only authorized entities can trigger these critical operations. Events from the contract:
+    - OracleAccountSet: Emitted when the Oracle account is set or updated.
+    - CommitmentProcessed: Emitted when a commitment has been processed by the Oracle, indicating whether it should be rewarded or slashed.
+
+- **Block Tracker**
+
+  This contract tracks Ethereum blocks and the winners of those blocks. The data collected here is crucial for determining the outcomes of commitments and deciding on rewards or   slashes. Access to the functions on this contract is also restricted to the Oracle key owner. Events from the contract:
+    - NewL1Block: Emitted when a new Layer 1 block is detected and tracked.
+    - NewWindow: Emitted when a new tracking window (a specific period of blocks) is initiated.
+    - OracleAccountSet: Emitted when the Oracle account responsible for this contract is set or updated.
+
+- **Validator Registry**
```

Comment:
I like the content you've added 👍, but there is more than one validator registry contract (see https://github.com/primev/mev-commit/tree/main/contracts/contracts/validator-registry) and this should be its own section (separate from core contracts). 

Either "validator registry" or "L1 contracts" would be appropriate for this section